src/backend/joanie/core/models/products.py

```diff
@@ -138,6 +145,43 @@ def __str__(self):
         return f"{self.product}: {self.position} / {self.course}"
 
 
+def on_add_course_runs_to_product_course_relation(action, instance, pk_set, **kwargs):
+    """
+    Signal triggered when course runs are added to a product course relation.
+    Some checks are processed before course runs are linked to product course relation.
+    """
+    if action != "pre_add":
+        return
+
+    if isinstance(instance, ProductCourseRelation):
+        relations = [instance]
+        course_runs_set = pk_set
+    else:
+        relations = ProductCourseRelation.objects.filter(pk__in=pk_set).select_related(
+            "course__course_runs"
+        )
+        course_runs_set = {instance.pk}
+
+    for relation in relations:
+        # Check that course runs relies on the relation course
+        if relation.course.course_runs.filter(pk__in=course_runs_set).count() != len(
+            course_runs_set
+        ):
+            raise ValidationError(
+                {
+                    "course_runs": [
+                        "Course runs to link does not relies on the relation course."
```

Comment:
Maybe clearer with something like:
```suggestion
                        "Limiting a course to targeted course runs can only be done for course runs already belonging to this course."
```